ql/src/java/org/apache/hadoop/hive/ql/io/HiveSequenceFileInputFormat.java

```diff
@@ -44,20 +44,20 @@ public HiveSequenceFileInputFormat() {
     setMinSplitSize(SequenceFile.SYNC_INTERVAL);
   }
 
-  private Set<FileStatus> fileStatuses = null;
+  private ThreadLocal<Set<FileStatus>> fileStatuses = new ThreadLocal<>();
 
   public void setFiles(Set<FileStatus> fileStatuses) {
-    this.fileStatuses= fileStatuses;
+    this.fileStatuses.set(fileStatuses);
   }
 
   @Override
   protected FileStatus[] listStatus(JobConf job) throws IOException {
-    if(fileStatuses== null || fileStatuses.isEmpty()) {
+    if(fileStatuses.get() == null || fileStatuses.get().isEmpty()) {
       // In cases where list of files to fetch is not provided we will use SequenceFileInputFormat
       // e.g. SELECT without a job
       return super.listStatus(job);
     }
-    FileStatus[] fsStatusArray = new FileStatus[fileStatuses.size()];
-    return fileStatuses.toArray(fsStatusArray);
+    FileStatus[] fsStatusArray = new FileStatus[fileStatuses.get().size()];
+    return fileStatuses.get().toArray(fsStatusArray);
```

Comment:
could be simplified to `fileStatuses.get().toArray(new FileStatus[0])`

Reply:
Done.